packages/runtime-common/realm.ts

```diff
@@ -3102,6 +3118,101 @@ export class Realm {
     return realmInfo;
   }
 
+  private async patchRealmInfo(
+    request: Request,
+    requestContext: RequestContext,
+  ): Promise<Response> {
+    let json: {
+      data?: { attributes?: { property?: string; value?: unknown } };
+    };
+    try {
+      json = await request.json();
+    } catch (e: any) {
+      return badRequest({
+        message: `The request body was not json: ${e.message}`,
+        requestContext,
+      });
+    }
+
+    let { property, value } = json.data?.attributes ?? {};
+    if (!property || typeof property !== 'string') {
+      return badRequest({
+        message: `The request body was missing a property name to update`,
+        requestContext,
+      });
+    }
+
+    let validators: Record<string, (value: unknown) => string | undefined> = {
+      name: (val) =>
+        typeof val === 'string' ? undefined : 'name must be a string',
+      backgroundURL: (val) =>
+        val === null || typeof val === 'string'
+          ? undefined
+          : 'backgroundURL must be a string or null',
+      iconURL: (val) =>
+        val === null || typeof val === 'string'
+          ? undefined
+          : 'iconURL must be a string or null',
+      showAsCatalog: (val) =>
+        val === null || typeof val === 'boolean'
+          ? undefined
+          : 'showAsCatalog must be a boolean or null',
+      interactHome: (val) =>
+        val === null || typeof val === 'string'
+          ? undefined
+          : 'interactHome must be a string or null',
+      hostHome: (val) =>
+        val === null || typeof val === 'string'
+          ? undefined
+          : 'hostHome must be a string or null',
+      publishable: (val) =>
+        val === null || typeof val === 'boolean'
+          ? undefined
+          : 'publishable must be a boolean or null',
+      visibility: (val) =>
+        val === 'private' || val === 'shared' || val === 'public'
+          ? undefined
+          : "visibility must be 'private', 'shared', or 'public'",
+    };
```

Comment:
This is readable but we’re adding enough validation that I wonder if we should be looking at Zod or the like